update.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+#! Exit immediately if any command fails
+set -e
+
+#! Define the repository URL and target directory
+repo_url="https://github.com/Xgorobot/RaspberryPi-CM4.git"
+target_dir="/home/pi/RaspberryPi-CM4-main"
+rc_local_file="/etc/rc.local"
+startup_command="su pi -c 'bash /home/pi/start1.sh'"
+startup_file="/home/pi/start1.sh"
+
+#! Check if the startup command is in /etc/rc.local, and add it if it's not
+if ! grep -qF "$startup_command" "$rc_local_file"; then
+Sudo sed -i "/^exit 0/i $startup_command" "$rc_local_file"
+fi
+
+if [ ! -f "$startup_file" ]; then
+echo "$startup_file"
+echo "cd /home/pi/RaspberryPi-CM4-main/" > "$startup_file"
+echo "sudo python3 main.py" >> "$startup_file"
+fi
+
+git config --global init.defaultBranch main
+git config --global user.email "xgo@update.com"
+
+if ! command -v git &> /dev/null; then
+echo "Git is not installed. Please install Git and try again."
+exit 1
+fi
+
+#! Check if the target directory exists
+if [ ! -d "$target_dir" ]; then
+# If it doesn't exist, clone the repository
+git clone "$repo_url" "$target_dir"
+elif [ ! -d "$target_dir/.git" ]; then
+# if the directory exist but not a git clone
+cd "$target_dir"
+echo "init"
+git init
+git checkout -b temp-branch
+git add .
+git commit -m 'importing local files'
+git remote add main https://github.com/Xgorobot/RaspberryPi-CM4.git
```

Comment:
Why not just remove the previously cloned repository and clone (preferably shallow) it again?

Reply:
Basically, the update of RPi-CM4 will take 2 lines of the code:
```sh
rm -rf ~/RaspberryPi-CM4
git clone --depth 1 --branch main https://github.com/Xgorobot/RaspberryPi-CM4.git ~/RaspberryPi-CM4
```

This saves roughly 200Mb of disk space.

Reply:
I copy the old faces in and have different test files which I didn't want to overwrite. I did see how it used the extra space but could not find a way to fix that. Is there a way to do both?

Reply:
Just keep your development files out from "power users" supposed to use the script?

If you decide to use some `~/develop/RaspberryPi-CM4` directory, you are certainly aware of how to use git, etc. fluently.

Reply:
Not sure what you are saying...

Reply:
used that --depth 1 switch and got it go to from 200mb on disk to 30mb...